Imaging schema draft

chili-chiu opened the pull request:
Imaging schema draft based on the [working group's doc](https://docs.google.com/document/d/1v34rJ2bGqF_6BUMLd2zvu36AReTmRu0dIxMCNhyqTYs/edit?tab=t.0).

chili-chiu committed:
add draft imaging schema

chili-chiu committed:
Update imaging_schema.md

incorporated Mikala's feedback

chili-chiu committed:
Update imaging_schema.md

chili-chiu committed:
Update imaging_schema.md

chili-chiu committed:
Update imaging_schema.md

chili-chiu committed:
Update imaging_schema.md

change instrument_version to SHOULD

chili-chiu committed:
Update imaging_schema.md

chili-chiu committed:
Update imaging_schema.md

chili-chiu committed:
Update imaging_schema.md

- changed to root level
- updated subject_procedure and specimen_procedure descriptions
- updated channel description: only applicable when channels exist
- removed acquisition_software and added acquisition_description

jrswedlow reviewed (commented):
Great work to this point @chili-chiu, and very useful to have this in this form.  A few comments:

- In general, we should formalize data structures and adoption of existing standards.  Examples are "Creation Time".  Consider adoption of ISO standard (e.g., https://en.wikipedia.org/wiki/ISO_8601), IUPAC  or similar.  
- Data Description contains some items that are free text (Acknowledgments), some items that are structured ("Author"; there are many ontologies that include this, but maybe we include name and ORCiD?), and items that probably should be standalone and structured ("License", with choices from [CC license set](https://creativecommons.org/licenses/list.en)?). Perhaps we do an experiment and align this info with [REMBI Study](https://www.ebi.ac.uk/bioimage-archive/rembi-model-reference/)?
- Label: we need to be careful here, as the semantics of this term are already defined in different ways in the OME-Zarr spec (https://ngff.openmicroscopy.org/0.5/index.html#labels-md) and has a completely different meaning.  From experience, I've often seen this concept as "channel". Capturing names of fluorescence channels is super complex ("Blue", "DAPI", "460nm", "DNA" are all valid and in use), and the various proposals are heavily dominated by the specific use cases. Perhaps contact Beth Cimini as she is working on this for an OME-Zarr RFC.  Note that the "omero specification"is heavily focused on rendering settings as that is a keep part of usability-- user sees usable image when they open it in a viewer).  For a different view, see https://github.com/miti-consortium/MITI.  
- For Specimen, would we propose that team use [protocols.io](https://protocols.io/)?  Probably a step too far.

chili-chiu commented:
- Agreed. 
- Data Description feedback may need to be surfaced to the team working on cross-modality specs. cc @brianraymor @dsadgat to help direct to the right owner there.
- Thanks for pointing that out! Yes we should avoid term collision. Propose changing "label" to "sample_label" to differentiate the usage. We want to separate "channel" from "sample_label" because in many cases the relationship is not one to one, and "channel" involves instrument and/or post-processing, while "sample_label" involves sample preparation. 
- For version 1.0.0, should we keep it light-weight? Let's evaluate protocols.io for future revisions?

chili-chiu commented:
> * Agreed.
> * Data Description feedback may need to be surfaced to the team working on cross-modality specs. cc @brianraymor @dsadgat to help direct to the right owner there.
> * Thanks for pointing that out! Yes we should avoid term collision. Propose changing "label" to "sample_label" to differentiate the usage. We want to separate "channel" from "sample_label" because in many cases the relationship is not one to one, and "channel" involves instrument and/or post-processing, while "sample_label" involves sample preparation.

wg feedback: Recommend to use "probe"
> * For version 1.0.0, should we keep it light-weight? Let's evaluate protocols.io for future revisions?

wg feedback: agreed that this is out of scope for version 1.0.0 and will evaluate for future iterations.

chili-chiu committed:
Rename schema.md to XMS_mapping.md

chili-chiu committed:
Update imaging_schema.md

- Updated metadata location
- Specified datetime to ISO 8601
- Changed metadata field name "Label" to "Probe"

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+Metadata describing the subject being imaged: Species, genotype, age, sex, and source.
+
+Cross-modality metadata: 
+[Cross-modality schema metadata](https://github.com/chanzuckerberg/data-guidance/tree/main/standards/cross-modality) are captured here.
```
This must reference a specific version of the cross-modality schema.

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+    <tr>
+      <th>Value</th>
+        <td><code>List[String]</code>. 
+         The List element MUST be a term in <a href="https://webprotege.stanford.edu/#projects/ee9f28eb-bcfe-469d-a9b4-48929f100eec/edit/Classes">the imaging methods list </a>.
```
This link prompted me for a login.

chili-chiu replied:
Will update to this doc link:
https://docs.google.com/document/d/1v34rJ2bGqF_6BUMLd2zvu36AReTmRu0dIxMCNhyqTYs/edit?tab=t.iuu4anexqv55#heading=h.l62d2b5ana

brianraymor replied:
Is the gdoc accessible to external readers?

chili-chiu replied:
Is there a GDrive location that can be accessible to external readers? 
I can move the doc to the suggested location.

brianraymor replied:
> Is there a GDrive location that can be accessible to external readers?

Unsure since the policies have changed over the years. CELLxGENE has a PII gdoc link that is supposed to be accessible when it was created by Legal at the time.  

That said, why not "copy as markdown" from the gdoc and inline into the requirement?

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+**For CryoET data:**
+- The full set of metadata for CryoET data is described [here](TBD). <br>
+
+The mapping of CryoET data and DCA data to the XMS is specified [here](https://github.com/chanzuckerberg/data-guidance/blob/main/standards/imaging/1.0.0/XMS_mapping.md).
```
"XMS" is an internal acronym which is not my cup of tea. It's not defined before its use here. I would always reference as the "cross-modality schema" for clarity. 

"DCA" is also not defined before its use?

chili-chiu replied:
Updated the terms with full names.

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+Administrative metadata about the source of the data, funding, relevant licenses, and restrictions on use.
+
+Cross-modality metadata: 
+Security & Compliance Metadata and Technical Metadata outlined in the [Data registration needs](https://czi.atlassian.net/wiki/spaces/SCI/pages/3896868957/Data+Registration+Needs+Metadata) are captured here.
```
I appreciate that there may be no other choice here at the moment, but I wouldn't expect this reference to be accessible to external readers.

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+
+
+## Overview
+The following descriptive metadata are associated with all imaging datasets to ensure that datasets can be found by searching for common characteristics of datasets. These metadata MUST be programmatically validated to ensure compliance. Additional metadata MAY be annotated at the discretion of data stewards.
```
It might be challenging to enforce all these metadata categories.

bentaculum replied:
Edit: Never mind, I am seeing below that there is a mixture of MUSTS and SHOULDS. I was under the impression that `MUST be programmatically validated` would enforce the existence of all fields.

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+
+**Categories of metadata:**
+- **Data description** — dataset-level metadata (source, license, compliance).  
+- **Subject** — species, genotype, sex, age.  
```
Usage of a specific cell line would be in `subject`, is that right?

chili-chiu replied:
Yes.

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+**For general images:**
+- All metadata are stored at the OME-Zarr root level. The metadata can be stored in separate json file(s) unless specified in OME-Zarr.
+
+**For high content screening (HCS):**
```
HCS guidance looks good to me

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+- **Channel description** — per-channel settings.  
+<br>
+
+For data in Zarr format, the metadata defined by [OME-Zarr](https://ngff.openmicroscopy.org/0.1/) SHOULD be implemented.<br>
```
This should probably refer to either `latest` OME-Zarr or alternatively fix the current version, 0.5.

uermel replied:
Unfortunately the cryoET data portal data is still in OME-Zarr v0.4 with no clear timeline for migration.

bentaculum replied:
Got it. The link here still needs fixing, it is pointing to v0.1.

chili-chiu replied:
Will update the link to the latest (https://ngff.openmicroscopy.org/latest/index.html).

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+- **Channel description** — per-channel settings.  
+<br>
+
+For data in Zarr format, the metadata defined by [OME-Zarr](https://ngff.openmicroscopy.org/0.1/) SHOULD be implemented.<br>
```
How about enforcing the minimal version of the OME-zarr format?

bentaculum commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+- Well-level metadata: subject, procedure, channel description.
+
+**For CryoET data:**
+- The full set of metadata for CryoET data is described [here](TBD). <br>
```
Might be worthwhile to point out the Dynamic Cell Atlas spec as well here https://chanzuckerberg.github.io/dynamic-cell-atlas-specs/

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+    <tr>
+      <th>Value</th>
+        <td><code>List[String]</code>. 
+         Examples: Zarr_V2, Zarr_V3, mrc.
```
I question the value of this field since I'm allowed to specify "foo", "bar", and "baz". Or "ZARR_V2" (unless normalized to lowercase) or "ZARRRR_V3".  How is this field used?

brianraymor replied:
Is it possible for validation/ingestion to introspect and detect?

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+- Well-level metadata: subject, procedure, channel description.
+
+**For CryoET data:**
+- The full set of metadata for CryoET data is described [here](TBD). <br>
```
TBD?

chili-chiu replied:
@uermel could you help to provide the link? Thanks.

dgmccart replied:
Should we link to this: [https://chanzuckerberg.github.io/cryoet-data-portal/stable/contributing_data.html#contributing-data](https://chanzuckerberg.github.io/cryoet-data-portal/stable/contributing_data.html#contributing-data)?

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+    </tr>
+    <tr>
+      <th>Annotator</th>
+      <td>OPTIONAL.</td>
```
Can you say more about the purpose of this field. In standards work, generally OPTIONAL is avoided because it does not get supported. (At MSFT, I was allowed to ignore all OPTIONAL requirements without requesting an exception when implementing a standard). Further, it's a string with no guidance.

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+</tbody></table>
+<br>
+
+## Procedure
```
I'd note that there's been conversations about similar metadata for chemical perturbations. See https://docs.google.com/document/d/1gRhhMdLaxaTilSUXwFPNxuul-fpW7ur9ztz4KPPdCP8/edit?pli=1&disco=AAABrkvLRSw and there was consensus that CELLxGENE would not capture because such information cannot be standardized/validated and would not be comparable between experiments.

brianraymor replied:
@BAevermann @jahilton ^^^ - do you wish to comment?

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+    </tr>
+    <tr>
+      <th>Value</th>
+        <td><code>List[String]</code>. 
```
Is there no further requirement for identifiers to "Proteins or other biomolecules"?  How is this validated?

brianraymor commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,402 @@
+    </tr>
+    <tr>
+      <th>Description</th>
+      <td>What this channel is intended to measure.</td>
```
What are the options? Is there a controlled vocabulary or related ontology?

chili-chiu committed:
Update imaging_schema.md

- Added DCA link
- Updated cross-modality schema link and specified version 1.1.0
- Updated OME-Zarr link
- Updated imaging_method link

ieivanov commented:
On the issue of "label" vs "probe" maybe I'll suggest yet another alternative - how about "marker"? I'm OK with either of the 3 as long as the definition of the term is clear.

I still think there is a need and opportunity to better link labels/probes/markers and channels as I suggested in #20 

I'd also like to hear the group's opinion on #19 - maybe we can bring it up for discussion at a future meeting?

dgmccart commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,403 @@
+
+**For specific imaging modalities and projects:**
+- The full set of metadata for CryoET data is described [here](TBD). <br>
+- The Dynamic Cell Atlas (DCA) project data spec is desbribed [here](https://chanzuckerberg.github.io/dynamic-cell-atlas-specs/). <br>
```
described typo

dgmccart commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,403 @@
+    <tr>
+      <th>Value</th>
+        <td><code>List[String]</code>. 
+         The List element MUST be a term in <a href="[https://tinyurl.com/63p8k9kf]">the imaging methods list </a>.
```
Hyperlink is not available in the rendered doc.

dgmccart replied:
Removing the brackets should fix it.

`The List element MUST be a term in <a href="https://tinyurl.com/63p8k9kf">the imaging methods list</a>.`

dgmccart commented on standards/imaging/1.0.0/imaging_schema.md:
```diff
@@ -0,0 +1,403 @@
+## Subject
+Metadata describing the subject being imaged: Species, genotype, age, sex, and source.
+
+Cross-modality metadata: 
```
Adding a sentence for context may better explain why we are linking to this. e.g. 

> Most information related to the subject will be described with the cross modality metadata. Refer to [v1.1.0 of the cross modality schema](https://github.com/chanzuckerberg/data-guidance/tree/main/standards/cross-modality/1.1.0) for details.
